src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java

```diff
@@ -120,10 +134,14 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
       }
 
       for (Peer peer : peers) {
-        Set<String> channels = hfClientWrapper.getHfClient().queryChannels(peer);
-        if (!channels.contains(channelName)) {
-          channel.addOrderer(orderer);
-          channel.joinPeer(peer);
+        if (peer.getProperties()
```

Comment:
If it is a peer from unexpected MSP, can also throw an exception or return an error. The error status can be unauthorized.